packages/gatsby/src/internal-plugins/query-runner/query-compiler.js

```diff
@@ -125,7 +126,12 @@ class Runner {
     let compilerContext = new GraphQLCompilerContext(this.schema)
     try {
       compilerContext = compilerContext.addAll(
-        ASTConvert.convertASTDocuments(this.schema, documents, validationRules)
+        ASTConvert.convertASTDocuments(
+          this.schema,
+          documents,
+          validationRules,
+          RelayParser.transform.bind(RelayParser)
```

Comment:
actually I think we should be using GraphQLParser ultimately...that should have less Relay specific validations and reqs. I'm not sure what they are yet, but should free us up some on the unnecessary validations

Reply:
What is this parser? We want a transformer though not a parser? I'm just copy and pasting though so not entirely sure what's going on :-)

Reply:
yeah it's super confusing, I'm not sure why it's called a Parser, but this logic used to just sit in ASTConvert, and turns the graphql ast into some IR format. Originally it just structured to turn it into just RelayIR, but in 1.4.x they split it out so the basic IR format is Relay agnostic (sort of) and then on top of that you can plug in this transformer to add more Relay specific stuff to the output...

tl;dr; is that `RelayParser` just extends `GraphQLParser` and adds some relay specific IR additions. This change is more correct as a migratiojn to the new version but ultimately we want to make use of the relay agnostic stuff they are splitting out

Reply:
Ah cool. So we're still on the status quo but at some point we could/should move.